pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -204,3 +251,92 @@ func (a *AWSActuator) updateProviderConfig(machineSet *machineapi.MachineSet, in
 		Value: &runtime.RawExtension{Object: providerConfig},
 	}
 }
+
+// getSubnetsByAvailabilityZones maps availability zones to subnet
+func (a *AWSActuator) getSubnetsByAvailabilityZone(pool *hivev1.MachinePool) (map[string]string, error) {
+	subnetsByAvailabilityZone := map[string]string{}
+	idPointers := make([]*string, len(pool.Spec.Platform.AWS.Subnets))
+	for i, id := range pool.Spec.Platform.AWS.Subnets {
+		idPointers[i] = aws.String(id)
+	}
+
+	results, err := a.awsClient.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: idPointers})
+	if err != nil {
+		conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+			pool.Status.Conditions,
+			hivev1.InvalidSubnetsMachinePoolCondition,
+			corev1.ConditionTrue,
+			"SubnetsNotFound",
+			err.Error(),
+			controllerutils.UpdateConditionIfReasonOrMessageChange,
+		)
+		if changed {
+			pool.Status.Conditions = conds
+			if err := a.client.Status().Update(context.Background(), pool); err != nil {
+				return nil, err
+			}
+		}
+		return nil, err
+	}
+
+	conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+		pool.Status.Conditions,
+		hivev1.InvalidSubnetsMachinePoolCondition,
+		corev1.ConditionFalse,
+		"SubnetsExist",
+		"All the subnets exist",
+		controllerutils.UpdateConditionNever,
+	)
+	if changed {
+		pool.Status.Conditions = conds
+		if err := a.client.Status().Update(context.Background(), pool); err != nil {
+			return nil, err
+		}
+	}
+
+	var zonesWithMultipleSubnets []string
+	for _, subnet := range results.Subnets {
+		_, ok := subnetsByAvailabilityZone[*subnet.AvailabilityZone]
+		if ok {
+			zonesWithMultipleSubnets = append(zonesWithMultipleSubnets, *subnet.AvailabilityZone)
```

Comment:
Use `sets.String` to accumulate the conflicts. Otherwise, if there are multiple conflicts, then you could end up with the same zone listed multiple times.

https://github.com/kubernetes/apimachinery/blob/bdcc9f4ab675d837965d1bd5babf8299afead78a/pkg/util/sets/string.go#L27.